### PR TITLE
feat: use debug logging when in debug mode (#128)

### PR DESCRIPTION
**Pull request by xremming**
Not 100% sure how to test changes easily. Decided to also include a few small cleanups of the code in the same PR.

After this PR, re-running a failed job with debugging will cause `mise-action` to always output logging at debug level which in turn should help debugging issues that happened in CI (motivated by my Friday debugging of [jdx/mise#2725](https://github.com/jdx/mise/issues/2725)).

Fixes: #128

### CODE REVIEW

**Comment by xremming** [comment]
@jdx Anything needed from me to get this merged?

File: src/index.ts [hunk 1]
```diff
@@ -89,7 +94,7 @@ async function restoreMiseCache(): Promise<void> {
   core.info(`mise cache restored from key: ${cacheKey}`)
 }
 
-async function setupMise(version: string | undefined): Promise<void> {
```
Comment (jdx) [inline src/index.ts:92]: isn't this usually set to `undefined`?

**Review by jdx — CHANGES_REQUESTED** [review]

**Comment by jdx** [comment]
I ran it with debug enabled and it works: https://github.com/jdx/mise-action/actions/runs/11326474236?pr=129

File: src/index.ts [hunk 2]
```diff
@@ -40,8 +40,13 @@ async function setEnvVars(): Promise<void> {
...
   }
-  if (getExperimental()) set('MISE_EXPERIMENTAL', '1')
-  set('MISE_LOG_LEVEL', core.getInput('log_level') || 'info')
+  if (core.getBooleanInput('experimental')) set('MISE_EXPERIMENTAL', '1')
+
+  const logLevel = core.isDebug()
+    ? 'debug'
+    : core.getInput('log_level') || 'info'
+
+  set('MISE_LOG_LEVEL', logLevel)
```
Comment (jdx) [inline src/index.ts:49]: if the user didn't specify debug mode or input log_level we should not set `MISE_LOG_LEVEL` at all

**Comment by jdx** [comment]
so I think at least for now we should only set the log level inside of our action, not set it for future actions. Otherwise I think it would add too much noise when doing unrelated things that just happen to call mise shims for example.

File: src/index.ts (hunk 1 above)
Comment (xremming) [reply src/index.ts:92]: I thought so too originally but the `core.getInput` actually returns an empty string when the value for the given variable is not set. This passes the type checks.

File: src/index.ts (hunk 2 above)
Comment (xremming) [reply src/index.ts:49]: The old (current) code defaults to info when debug level is not set so I kept that behaviour. I agree that maybe it makes more sense to not set the `MISE_LOG_LEVEL` when the user has not set `log_level`.

I'll make that change.

**Comment by jdx** [comment]
> Hmm. Okay I think I understand what you mean. This PR doesn't change the current behaviour which is to export the log level to all further steps too.

that is fine in the scenario where a user explicitly sets log_level. It's not ok for us to set it automatically just because debugging was enabled on the action generally. I think it's fine within our step but it would create a lot of noise that is likely not what the user would want in unrelated steps.

File: src/index.ts [hunk 3]
```diff
@@ -151,10 +149,17 @@ const mise = async (args: string[]): Promise<number> =>
       core.getInput('working_directory') ||
       core.getInput('install_dir') ||
       process.cwd()
+    const env = core.isDebug()
+      ? { ...process.env, MISE_LOG_LEVEL: 'debug' }
```
Comment (jdx) [inline src/index.ts:153]:
```suggestion
      ? { MISE_LOG_LEVEL: 'debug', ...process.env }
```

if the log level is already set (say, to "trace") we should respect that

Comment (xremming) [reply src/index.ts]: Amended the fix to the commit.

**Comment by xremming** [comment]
Looks like you just fixed the cache_save bug that I had fixed in the branch so I rebased and dropped the commit (and squashed some of my changes). Let me know if you want me to rebase again and clean the commits.

**Pull request merged**